public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -255,4 +256,10 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
   getSubscriptions() {
     return this.azureMonitorDatasource.getSubscriptions();
   }
+
+  interpolateVariablesInQueries(queries: AzureMonitorQuery[], scopedVars: ScopedVars): AzureMonitorQuery[] {
+    return queries.map(
```

Comment:
You have checked the queries length in other data sources but not here. Shouldn't you check it?

Reply:
The reason is that both in this and in azure I'm only reusing pre-existing logic to interpolate variables

Reply:
To be more specific:

while in the other datasources the fields to be interpolated are directly defined in `interpolateVariablesInQueries`, therefore the test and the implementation are coupled, here I'm using pre-existing logic that is not really tied to this function. 

Testing for how many fields get interpolated here means that whenever we change `applyTemplateVariables` in this case or `prepareTimeSeriesQuery` in cloud-monitoring we are going to break tests that are not coupled with the implementation.

I'm not really 100% happy with the tests anyway, so if you have suggestions I would really appreciate it!